addons/l10n_sa/models/account_move.py

```diff
@@ -2,20 +2,24 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 import base64
 
-from odoo import api, fields, models, _
+from odoo import _, api, fields, models
 from odoo.exceptions import UserError
-from odoo.tools import float_repr
+from odoo.tools import float_repr, format_datetime
 
 
 class AccountMove(models.Model):
     _inherit = 'account.move'
 
-    l10n_sa_delivery_date = fields.Date(string='Delivery Date', default=fields.Date.context_today, copy=False,
+    l10n_sa_delivery_date = fields.Date(string='Supply Date', default=fields.Date.context_today, copy=False,
                                         readonly=True, states={'draft': [('readonly', False)]},
-                                        help="In case of multiple deliveries, you should take the date of the latest one. ")
+                                        help="""Date when the supply of the goods and services is performed.
+                                        You can set this date to when you want ZATCA to recognize the VAT Due Liability as ZATCA will consider the earlier of this date and the Issue Date.
+                                        In case of multiple deliveries, you should take the date of the latest one.""")
     l10n_sa_show_delivery_date = fields.Boolean(compute='_compute_show_delivery_date')
     l10n_sa_qr_code_str = fields.Char(string='Zatka QR Code', compute='_compute_qr_code_str')
-    l10n_sa_confirmation_datetime = fields.Datetime(string='Confirmation Date', readonly=True, copy=False)
+    l10n_sa_confirmation_datetime = fields.Datetime(string='Issue Date', readonly=True, copy=False,
+                                                    help="""Date when the invoice is confirmed and posted.
+                                                    In other words, it is the date on which the invoice is generated as final document (after securing all internal approvals).""")
```

Comment:
```suggestion
    l10n_sa_confirmation_datetime = fields.Datetime(
        string='Issue Date',
        readonly=True,
        copy=False,
        help="""Date when the invoice is confirmed and posted.
        In other words, it is the date on which the invoice is generated as final document (after securing all internal approvals)."""
    )
```

I would prefer the same above for l10n_sa_delivery_date, but a bit of your choice.
Be careful that you will need to update the pot with the change, if I'm not mistaken

PS: I don't really mind this if that's all there was in the PR, but as I have other things to say, I'm asking you :)